src/goto-instrument/contracts/dynamic-frames/dfcc_swap_and_wrap.cpp

```diff
@@ -57,38 +56,45 @@ dfcc_swap_and_wrapt::dfcc_swap_and_wrapt(
 }
 
 // static map
-std::map<irep_idt, std::pair<irep_idt, dfcc_contract_modet>>
+std::map<
+  irep_idt,
+  std::pair<irep_idt, std::pair<dfcc_contract_modet, dfcc_loop_contract_modet>>>
   dfcc_swap_and_wrapt::cache;
 
 void dfcc_swap_and_wrapt::swap_and_wrap(
   const dfcc_contract_modet contract_mode,
+  const dfcc_loop_contract_modet loop_contract_mode,
   const irep_idt &function_id,
   const irep_idt &contract_id,
   std::set<irep_idt> &function_pointer_contracts,
   bool allow_recursive_calls)
 {
-  auto pair = cache.insert({function_id, {contract_id, contract_mode}});
+  auto pair = cache.insert(
+    {function_id, {contract_id, {contract_mode, loop_contract_mode}}});
   auto inserted = pair.second;
 
   if(!inserted)
   {
-    auto old_contract_id = pair.first->second.first;
-    auto old_contract_mode = pair.first->second.second;
+    irep_idt old_contract_id = pair.first->second.first;
+    dfcc_contract_modet old_contract_mode = pair.first->second.second.first;
+    dfcc_loop_contract_modet old_loop_contract_mode =
+      pair.first->second.second.second;
```

Comment:
Could we create regression test to cover the lines below?

Reply:
this should already be covered as soon as we enforce a contract or replace a contract, are you sure this is not already covered ?

Reply:
[Codecov](https://github.com/diffblue/cbmc/pull/7541/checks?check_run_id=13347318605) info says it's not covered. False alarm?